include/osquery/extensions.h

```diff
@@ -50,7 +50,7 @@ Status queryExternal(const std::string& query, QueryData& results);
 Status getQueryColumnsExternal(const std::string& q, TableColumns& columns);
 
 /// External (extensions) SQL implementation plugin provider for "sql" registry.
-class ExternalSQLPlugin : SQLPlugin {
+class ExternalSQLPlugin : public SQLPlugin {
```

Comment:
NB: I am extremely surprised this used to work. Perhaps our lax use of `Plugin` inheritance enabled it. We will now create a compiler error if something attempts to register a class that hasn't inherited from `Plugin`, meaning implementing a `call` method.

This is because we require *covariance* with our usable of shared pointers for the container of registered plugins.